File: SDK/CentralService/dotnet/src/CentralService.Client/Models/ServiceRegistrationRequest.cs
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace CentralService.Client.Models
{
    /// <summary>
    /// 表示服务注册请求。
    /// </summary>
    [DataContract]
    public sealed class ServiceRegistrationRequest
    {
        /// <summary>
        /// 获取或设置服务实例标识。
        /// </summary>
        [DataMember(Name = "id", Order = 1, EmitDefaultValue = false)]
        public string Id { get; set; }

        /// <summary>
        /// 获取或设置服务名称。
        /// </summary>
        [DataMember(Name = "name", Order = 2)]
        public string Name { get; set; }

        /// <summary>
        /// 获取或设置服务主机地址。
        /// </summary>
        [DataMember(Name = "host", Order = 3)]
        public string Host { get; set; }

        /// <summary>
        /// 获取或设置服务端口。
        /// </summary>
        [DataMember(Name = "port", Order = 4)]
        public int Port { get; set; }

        /// <summary>
        /// 获取或设置服务类型。
        /// </summary>
        [DataMember(Name = "serviceType", Order = 5, EmitDefaultValue = false)]
        public string ServiceType { get; set; }

        /// <summary>
        /// 获取或设置健康检查地址。
        /// </summary>
        [DataMember(Name = "healthCheckUrl", Order = 6, EmitDefaultValue = false)]
        public string HealthCheckUrl { get; set; }

        /// <summary>
        /// 获取或设置健康检查端口。
        /// </summary>
        [DataMember(Name = "healthCheckPort", Order = 7, EmitDefaultValue = false)]
        public int HealthCheckPort { get; set; }

        /// <summary>
        /// 获取或设置中心服务通过 WebSocket 向该服务发送心跳请求的频率（秒）。
        /// 为 0 时表示不发送心跳请求。
        /// </summary>
        [DataMember(Name = "heartbeatIntervalSeconds", Order = 8, EmitDefaultValue = false)]
        public int HeartbeatIntervalSeconds { get; set; }

        /// <summary>
        /// 获取或设置负载均衡权重。
        /// </summary>
        [DataMember(Name = "weight", Order = 9, EmitDefaultValue = false)]
        public int Weight { get; set; }

        /// <summary>
        /// 获取或设置附加元数据。
        /// </summary>
        [DataMember(Name = "metadata", Order = 10, EmitDefaultValue = false)]
        public Dictionary<string, string> Metadata { get; set; }
    }
}

File: SDK/CentralService/java/client/src/main/java/erp/centralservice/client/models/ServiceRegistrationRequest.java
package erp.centralservice.client.models;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * 服务注册请求模型。
 *
 * <p>字段名称与服务端 JSON 协议保持一一对应，新增字段前需要同时更新服务端契约。</p>
 */
public final class ServiceRegistrationRequest {
    /** 可选的服务 ID；为空时通常由服务端生成。 */
    public String id;
    /** 服务名称。 */
    public String name;
    /** 服务主机地址。 */
    public String host;
    /** 服务端口。 */
    public int port;
    /** 服务类型。 */
    public String serviceType;
    /** 健康检查地址。 */
    public String healthCheckUrl;
    /** 健康检查端口。 */
    public int healthCheckPort;
    /** 路由或调度权重。 */
    public int weight;
    /** 扩展元数据。 */
    public Map<String, String> metadata;

    /**
     * 转换为与中心服务接口兼容的 JSON 对象。
     *
     * @return 可直接序列化的键值映射
     */
    public Map<String, Object> toJson() {
        Map<String, Object> m = new LinkedHashMap<String, Object>();
        m.put("id", id);
        m.put("name", name);
        m.put("host", host);
        m.put("port", port);
        m.put("serviceType", serviceType);
        m.put("healthCheckUrl", healthCheckUrl);
        m.put("healthCheckPort", healthCheckPort);
        m.put("weight", weight);
        m.put("metadata", metadata);
        return m;
    }
}
